Docs: adding CDN solution to Quick start guide. (#11942)
ckeditor/ckeditor5: docs/installation/getting-started/quick-start.md

The plugins not available currently in the superbuild include:
* Watchdog
* ContextWatchdog
* Context
* Title

### Using the CKEditor 5 superbuild

To start using the CKEditor 5 superbuild from CDN, first add the editor placeholder to your document.

```html
<div id="editor"></div>
```

Then, include the code from CDN to superbuild to load the editor.

```html
<script src="https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/super-build/ckeditor.js"></script>
```

In the superbuild, all editor classes are stored under the `CKEDITOR` object. To create the classic editor, you need to access the `CKEDITOR` object first, then call the {@link module:editor-classic/classiceditor~ClassicEditor#create `ClassicEditor.create()`} method.

```html
<script>
	CKEDITOR.ClassicEditor
		.create( document.querySelector( '#editor' ) )
		.catch( error => {
			console.error( error );
		} );

> </script>

Reviewer: This section suggests that doing those steps is enough to make the editor work (which is true for the previous tutorial about using CDN, it is possible to just paste similar lines to an HTML document and it is working), for this guide it is not true and misleading. When you paste just those steps you start receiving errors about missing config of multiple plugins. e.g. RH. I see two options here: delete those steps and refer to full examples or make a disclaimer that due to the complex character of the superbuild there are more configurations to do which are explained in detail below. At this stage, it might be very confusing for readers who do not have experience with editors and want to give it a try. 

I would go with the attitude to include a full fully configured example of an editor without premium features and then an example with premium features. We should focus on quick experience first and then go deep into details.

				'resizeImage'
			],
			insert: {
				integrations: [
					'insertImageViaUrl'
				]
			}
		},
		list: {
			properties: {
				styles: true,
				startIndex: true,
				reversed: true
			}
		},
		link: {
			decorators: {
				addTargetToExternalLinks: true,
				defaultProtocol: 'https://',
				toggleDownloadable: {
					mode: 'manual',
					label: 'Downloadable',
					attributes: {
						download: 'file'
					}
				}
			}
		},
		mention: {
			feeds: [
				{
					marker: '@',
					feed: [
						'@apple', '@bears', '@brownie', '@cake', '@cake', '@candy', '@canes', '@chocolate', '@cookie', '@cotton', '@cream',
						'@cupcake', '@danish', '@donut', '@dragée', '@fruitcake', '@gingerbread', '@gummi', '@ice', '@jelly-o',
						'@liquorice', '@macaroon', '@marzipan', '@oat', '@pie', '@plum', '@pudding', '@sesame', '@snaps', '@soufflé',
						'@sugar', '@sweet', '@topping', '@wafer'
					],
					minimumCharacters: 1
				}
			]
		},
		placeholder: 'Type or paste your content here!',
		table: {
			contentToolbar: [
				'tableColumn', 'tableRow', 'mergeTableCells', 'tableProperties', 'tableCellProperties', 'toggleTableCaption'
			]
		},
	} )
	.catch( error => {
		console.log( error );
	} );

> ```

Reviewer: In general, I agree that step by step attitude and in the end presenting full implementation is good, but in this case, pasting a large amount of source code to say that plugins can be removed and just after that place another large amount of almost the same code with full implementation looks like too much. 
I would suggest removing this snippet and within the part about eliminating plugins and customising the toolbar refer to the full implementation below.

							editorContainer.style.display = 'none';

							const toolbarContainer = document.querySelector( '.editor-toolbar' );
							toolbarContainer.innerHTML = '';
							toolbarContainer.appendChild( viewerEditor.ui.view.toolbar.element );

							return viewerEditor;
						} );
					},
					closeRevisionViewerCallback: viewerEditor => {
						const editorContainer = document.querySelector( '#editor-container' );
						const viewerContainer = document.querySelector( '#revision-viewer-container' );

						viewerContainer.style.display = 'none';
						editorContainer.style.display = '';

						return viewerEditor.destroy().then( () => {
							const toolbarContainer = document.querySelector( '.editor-toolbar' );
							toolbarContainer.innerHTML = '';
							toolbarContainer.appendChild( window.editor.ui.view.toolbar.element );
						} );
					}
			},
	} )
	.then( editor => {
		console.log( 'Editor was initialized', editor );
		const toolbarContainer = document.querySelector( '.editor-toolbar' );

		toolbarContainer.appendChild( editor.ui.view.toolbar.element );
	} )
	.catch( error => {
		console.error( error.stack );
		console.log( error );
	} );
```

> You can now run webpack to build the application. To do that, call the `webpack` executable:

Reviewer: As above, the unnecessary huge amount of code and repetition of most of it just behind, I think we might delete this and refer to the main example.

				return CKEDITOR.DecoupledEditor.create( viewerElement, config ).then( viewerEditor => {
					viewerContainer.style.display = 'flex';
					editorContainer.style.display = 'none';

					const toolbarContainer = document.querySelector( '.editor-toolbar' );
					toolbarContainer.innerHTML = '';
					toolbarContainer.appendChild( viewerEditor.ui.view.toolbar.element );

					return viewerEditor;
				} );
			},
			closeRevisionViewerCallback: viewerEditor => {
				const editorContainer = document.querySelector( '#editor-container' );
				const viewerContainer = document.querySelector( '#revision-viewer-container' );

				viewerContainer.style.display = 'none';
				editorContainer.style.display = '';

				return viewerEditor.destroy().then( () => {
					const toolbarContainer = document.querySelector( '.editor-toolbar' );
					toolbarContainer.innerHTML = '';
					toolbarContainer.appendChild( window.editor.ui.view.toolbar.element );
				} );
			}
		},
	} )
	.then( editor => {
		const toolbarContainer = document.querySelector( '.editor-toolbar' );

		toolbarContainer.appendChild( editor.ui.view.toolbar.element );
	} )
	.catch( error => {
		console.log( error );
	} );

> ```

Reviewer: As above, the unnecessary huge amount of code and repetition of most of it just behind, I think we might delete this and refer to the main example explaining the details.

Author: @dufipl - please compare the solution we proposed with collapsible code listings.

<info-box>
	Please consider, that the superbuild contains a really whole lot of code. A good portion of that code may not be needed in you implementation, so using the superbuild should rather be considered for evaluation purposes and for tests, than for production environment.

	We strongly advise using the {@link installation/getting-started/quick-start-other#creating-custom-builds-with-online-builder Online builder} approach or {@link installation/getting-started/quick-start-other#building-the-editor-from-source building the editor from source} to create customized and efficient production-environment solutions. You can also try out the {@link installation/advanced/predefined-builds predefined builds} tailored for specific needs.
</info-box>

### Using the CKEditor 5 superbuild

First add the editor placeholder to your document.

```html
<div id="editor"></div>
```

Then include the code from CDN to superbuild to load the editor.

```html
<script src="https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/super-build/ckeditor.js"></script>
```

> Call the {@link module:editor-classic/classiceditor~ClassicEditor#create `ClassicEditor.create()`} method.

Reviewer 2: ```suggestion
In the superbuild, all editor classes are stored under the `CKEDITOR` object. To create the classic editor, you need to access `CKEDITOR` object first, then call the {@link module:editor-classic/classiceditor~ClassicEditor#create `ClassicEditor.create()`} method.
```

I've added a small suggestion to make it clear how to access editor classes in the superbuild.

```html
<script>
	CKEDITOR.ClassicEditor
		.create( document.querySelector( '#editor' ) )
		.catch( error => {
			console.error( error );
		} );
</script>
```

> Remove the plugins you do not need with the `removePlugins` method. In this example, we will remove the premium collaboration features that require credentials to work. We need to do this, otherwise the editor will throw an error.

Reviewer 2: ```suggestion
Remove the plugins you do not need with the `removePlugins` configuration option. In this example, we will remove the premium collaboration features that require credentials to work. We need to do this, otherwise the editor will throw an error.
```

Then, configure the toolbar to display only the desired options. You can read more about toolbar configuration the the {@link features/toolbar toolbar guide}. Several plugins, like the image feature or the list feature need additional configuration for their own toolbars.

A source code listing for the configuration can be seen below.

```js
CKEDITOR.ClassicEditor
	.create( document.querySelector( '#editor' ), {

> 		removePlugins: [ 'Comments, RealTimeCollaborativeComments, TrackChanges, RealTimeCollaborativeTrackChanges, RevisionHistory, RealTimeCollaborativeRevisionHistory, RealTimeCollaborativeEditing, ExportWord, ExportPdf, Pagination, WProofreader, MathType' ],

Reviewer 2: Each item should be wrapped in `''`. Also, we should add `PresenceList` and `TrackChangesData` to this list.

```suggestion
		removePlugins: [ 'Comments', 'TrackChanges', 'TrackChangesData', 'RevisionHistory', 'RealTimeCollaborativeComments', 'RealTimeCollaborativeTrackChanges', 'RealTimeCollaborativeRevisionHistory', 'RealTimeCollaborativeEditing', 'PresenceList', 'ExportWord', 'ExportPdf', 'Pagination', 'WProofreader', 'MathType' ],
```

		toolbar: {
			items: [
				'undo', 'redo',
				'|',

> 				'exportPdf', 'exportWord',

Reviewer 2: Exports are removed in this snippet, so the toolbar shouldn't contain items related to these plugins.

				'|',

> 				'wproofreader', 'findAndReplace', 'selectAll',

Author: Addressed in commit.

					marker: '@',
					feed: [
						'@apple', '@bears', '@brownie', '@cake', '@cake', '@candy', '@canes', '@chocolate', '@cookie', '@cotton', '@cream',
						'@cupcake', '@danish', '@donut', '@dragée', '@fruitcake', '@gingerbread', '@gummi', '@ice', '@jelly-o',
						'@liquorice', '@macaroon', '@marzipan', '@oat', '@pie', '@plum', '@pudding', '@sesame', '@snaps', '@soufflé',
						'@sugar', '@sweet', '@topping', '@wafer'
					],
					minimumCharacters: 1
				}
			]
		},
		placeholder: 'Type or paste your content here!',
		table: {
			contentToolbar: [
				'tableColumn', 'tableRow', 'mergeTableCells', 'tableProperties', 'tableCellProperties', 'toggleTableCaption'
			]
		},
	} )
	.catch( error => {
		console.log( error );
	} );
```

### Sample implementation

```html
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<title>CKEditor 5 – Full-featured editor</title>
	<script src="https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/super-build/ckeditor.js"></script>
</head>
<body>
	<h1>Full-featured editor</h1>
	<div id="editor">
		<p>This is some sample content.</p>
	</div>
	<script>
		CKEDITOR.ClassicEditor
			.create( document.querySelector( '#editor' ), {

> 				removePlugins: [ 'Comments, RealTimeCollaborativeComments, TrackChanges, RealTimeCollaborativeTrackChanges, RevisionHistory, RealTimeCollaborativeRevisionHistory, RealTimeCollaborativeEditing, ExportWord, ExportPdf, Pagination, WProofreader, MathType' ],

Reviewer 2: We should update this part. thesame way as above.

Author: Addressed in commit.

				toolbar: {
					items: [
						'undo', 'redo',
						'|',

> 						'exportPdf', 'exportWord',

Reviewer 2: We should update this part. thesame way as above.

Author: Aren't exports available in this version?

Author: Addressed in commit.

						'|',

> 						'wproofreader', 'findAndReplace', 'selectAll',

Reviewer 2: We should update this part same way as above.

Author: Addressed in commit.
